yq: add page

joshuatcasey opened the pull request:
- Include yq v4 and v3 examples

<!-- Thank you for sending a PR! -->
<!-- Please perform the following checks and mark all the boxes accordingly. -->
<!-- You can remove the checklist items that don't apply to your PR. -->

- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

navarroaxel commented on pages/common/yq.md:
```diff
@@ -0,0 +1,37 @@
+
+- Output a YAML file, in pretty-print format (yq v4+):
+
+`$ yq eval {{file.yaml}}`
```
The examples should not be with a prompt, and should use the `path/to/file`:
```suggestion
`yq eval {{path/to/file.yaml}}`
```

navarroaxel commented on pages/common/yq.md:
```diff
@@ -0,0 +1,37 @@
+
+> A lightweight and portable command-line YAML processor.
+> Note that `yq` v3 and v4 do not share commands.
+> More v3/v4 examples at <https://mikefarah.gitbook.io/yq/v/v4.x/upgrading-from-v3>.
```
```suggestion
> More information: <https://mikefarah.gitbook.io/yq/>.
```

navarroaxel commented on pages/common/yq.md:
```diff
@@ -0,0 +1,37 @@
+# yq
+
+> A lightweight and portable command-line YAML processor.
+> Note that `yq` v3 and v4 do not share commands.
```
We use to support only the latest version of a command. 
```suggestion
```

joshuatcasey replied:
I think it makes a lot of sense for `tldr` to support only the most recent release. `yq` 4 was release only 8 days ago and has a completely different UX than `yq` 3 so I felt it was important to keep the continuity. 

https://github.com/mikefarah/yq/releases/tag/4.0.0

bl-ue replied:
@navarroaxel are you okay with that?  If so go ahead and merge this.

einverne reviewed (approved):
LGTM.

sbrl commented on pages/common/yq.md:
```diff
@@ -0,0 +1,36 @@
+> A lightweight and portable command-line YAML processor.
+> More information: <https://mikefarah.gitbook.io/yq/>.
+
+- Output a YAML file, in pretty-print format (yq v4+):
```
Perhaps this will read better?
```suggestion
- Output a YAML file, in pretty-print format (`yq` v4+):
```

Same for the other instances.

bl-ue replied:
Or maybe just `v4+`...?

bl-ue replied:
@joshuatcasey @sbrl what do you think?

joshuatcasey replied:
I think simpler is better, so how about just "(v4+)".

bl-ue replied:
I agree.  Go for it,

sbrl replied:
Yeah, let's go with `(v4+)`.

joshuatcasey commented:
Since this branch has gotten a little stale and there's been a few tweaks I rebased-and-squashed all the feedback up to now.

Thanks for all the feedback @sbrl @bl-ue @einverne @navarroaxel @sahilister .

bl-ue commented:
@joshuatcasey try to avoid force-pushing when possible, because it can make reviewing a bit difficult and we squash commits into one when we merge.  Other than that...

bl-ue reviewed (approved):
Looks good 👍  Thank you very much @joshuatcasey!

bl-ue commented on pages/common/yq.md:
```diff
@@ -0,0 +1,36 @@
+
+`yq read {{path/to/file.yaml}} '[0]'`
+
+- Overwrite (or set) a key to a value in a file (v4+):
```
How about switching "overwrite" and "set", like this?
```suggestion
- Set (or overwrite) a key to a value in a file (v4+):
```

bl-ue commented on pages/common/yq.md:
```diff
@@ -0,0 +1,36 @@
+
+`yq eval '.{{key}} = "{{value}}"' --inplace {{path/to/file.yaml}}`
+
+- Overwrite (or set) a key to a value in a file (v3):
```
Same here.
```suggestion
- Set (or overwrite) a key to a value in a file (v3):
```

bl-ue commented on pages/common/yq.md:
```diff
@@ -0,0 +1,36 @@
+
+`yq eval-all 'select(filename == "{{path/to/file1.yaml}}") * select(filename == "{{path/to/file2.yaml}}")' {{path/to/file1.yaml}} {{path/to/file2.yaml}}`
+
+- Merge two files and print to STDOUT (v3):
```
```suggestion
- Merge two files and print to `stdout` (v3):
```

bl-ue commented on pages/common/yq.md:
```diff
@@ -0,0 +1,36 @@
+
+`yq write --inplace {{path/to/file.yaml}} '{{key}}' '{{value}}'`
+
+- Merge two files and print to STDOUT (v4+):
```
Lowercase + `code` is the most common formatting of `stdout`:
```suggestion
- Merge two files and print to `stdout` (v4+):
```

bl-ue reviewed (commented):
Found a few very minor things on second look...

sbrl reviewed (approved):
Thanks, @joshuatcasey!
